hack/tools/mdbook/releaselink/releaselink.go

```diff
@@ -0,0 +1,96 @@
+// +build tools
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/blang/semver"
+	"golang.org/x/tools/go/vcs"
+	"sigs.k8s.io/kubebuilder/docs/book/utils/plugin"
+)
+
+// ReleaseLink.
+type ReleaseLink struct{}
+
+func (_ ReleaseLink) SupportsOutput(_ string) bool { return true }
+func (l ReleaseLink) Process(input *plugin.Input) error {
+	return plugin.EachCommand(&input.Book, "releaselink", func(chapter *plugin.BookChapter, args string) (string, error) {
+		tags := reflect.StructTag(strings.TrimSpace(args))
+
+		gomodule := tags.Get("gomodule")
+		asset := tags.Get("asset")
+		versionRange := semver.MustParseRange(tags.Get("version"))
+
+		repo, err := vcs.RepoRootForImportPath(gomodule, false)
+		if err != nil {
+			return "", err
+		}
+
+		rawURL := url.URL{
+			Scheme: "https",
+			Host:   "proxy.golang.org",
+			Path:   path.Join(gomodule, "@v", "/list"),
+		}
+
+		resp, err := http.Get(rawURL.String())
```

Comment:
Where in the pipeline does release happen?
If it's inconvenient for the preprocessor to fail you might consider a retry here. But if it's not a big deal to simply re-run then this is not an issue

Reply:
If it fail, it should be ok to just retry, it doesn't take long in CI (<1m) or locally (<2s)